azure-configurationservice/tests/test_azure_configuration_client.py

```diff
@@ -0,0 +1,601 @@
+            tags=tags,
+            etag=to_update_kv.etag,
+        )
+        # remove the new key value from the service
+        self.app_config_client.delete_configuration_setting(
+            updated_kv.key, label=updated_kv.label
+        )
+        assert (
+            to_update_kv.key == updated_kv.key
+            and to_update_kv.label == updated_kv.label
+            and "updated_value" == updated_kv.value
+            and to_update_kv.content_type == updated_kv.content_type
+            and tags == updated_kv.tags
+            and to_update_kv.etag != updated_kv.etag
+        )
+
+    def test_update_existing_configuration_setting_label_noetag(self):
+        sample_kv = copy(self.test_data.label1_data[-1])
+
+        # create a new key value into AzConfig service
+        sample_kv.key = "unit_test_key_" + self.test_data.key_uuid
+        to_update_kv = self.app_config_client.add_configuration_setting(sample_kv)
+        tags = {"a": "b", "c": "d"}
+        updated_kv = self.app_config_client.update_configuration_setting(
+            to_update_kv.key, label=to_update_kv.label, value="updated_value", tags=tags
+        )
+        # remove the new key value from the service
+        self.app_config_client.delete_configuration_setting(
+            updated_kv.key, label=updated_kv.label
+        )
+        assert (
+            to_update_kv.key == updated_kv.key
+            and to_update_kv.label == updated_kv.label
+            and "updated_value" == updated_kv.value
+            and to_update_kv.content_type == updated_kv.content_type
+            and tags == updated_kv.tags
+            and to_update_kv.etag != updated_kv.etag
+        )
+
+    def test_update_existing_configuration_setting_label_wrong_etag(self):
+        sample_kv = copy(self.test_data.label1_data[-1])
+
+        # create a new key value into AzConfig service
+        sample_kv.key = "unit_test_key_" + self.test_data.key_uuid
+        to_update_kv = self.app_config_client.add_configuration_setting(sample_kv)
+        tags = {"a": "b", "c": "d"}
+        etag = "wrong etag"
+        with pytest.raises(CloudError):
+            self.app_config_client.update_configuration_setting(
+                to_update_kv.key,
+                label=to_update_kv.label,
+                value="updated_value",
+                tags=tags,
+                etag=etag,
+            )
+        self.app_config_client.delete_configuration_setting(
+            to_update_kv.key, label=to_update_kv.label
+        )
+
+    def test_update_no_existing_configuration_setting_label_noetag(self):
+        key = self.test_data.key_uuid
+        label = "test_label1"
+        with pytest.raises(CloudError):
+            self.app_config_client.update_configuration_setting(
+                key, label=label, value="some value"
+            )
+
+    # method: get_configuration_setting
+    def test_get_configuration_setting_no_label(self):
+        compare_kv = self.test_data.no_label_data[0]
```

Comment:
how do you guarantee that the setting is still available and no other process delete it?
It will be better if in the tests you create a setting and then you get it. that way you know it is there

Reply:
That's what the setUp( ) is for.